rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,241 @@
+from rasa.constants import (
+    CONFIG_AUTOCONFIGURABLE_KEYS,
+    DOCS_URL_PIPELINE,
+    DOCS_URL_POLICIES,
+    CONFIG_KEYS,
+)
+from rasa.utils import io as io_utils, common as common_utils
+
+logger = logging.getLogger(__name__)
+
+COMMENTS_FOR_KEYS = {
+    "pipeline": (
+        f"# # No configuration for the NLU pipeline was provided. The following "
+        f"default pipeline was used to train your model.\n"
+        f"# # To customise it, uncomment and adjust the pipeline. See "
+        f"{DOCS_URL_PIPELINE} for more information.\n"
+    ),
+    "policies": (
+        f"# # No configuration for policies was provided. The following default "
+        f"policies were used to train your model.\n"
+        f"# # To customise them, uncomment and adjust the policies. See "
+        f"{DOCS_URL_POLICIES} for more information.\n"
+    ),
+}
+
+
+def get_configuration(config_file_path: Text) -> Dict[Text, Any]:
+    """Determine configuration from a configuration file.
+
+    Keys that are provided and have a value in the file are kept. Keys that are not
+    provided are configured automatically.
+
+    Args:
+        config_file_path: The path to the configuration file.
+    """
+    if config_file_path and os.path.exists(config_file_path):
+        config = io_utils.read_config_file(config_file_path)
+
+        missing_keys = _get_missing_config_keys(config)
+        keys_to_configure = _get_unspecified_autoconfigurable_keys(config)
+
+        if keys_to_configure:
+            config = _auto_configure(config, keys_to_configure)
+            _dump_config(config, config_file_path, missing_keys, keys_to_configure)
+
+    else:
+        logger.debug("No configuration file was provided to the TrainingDataImporter.")
+        config = {}
+
+    return config
+
+
+def _get_unspecified_autoconfigurable_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+
+
+def _get_missing_config_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_KEYS if k not in config.keys()]
+
+
+def _auto_configure(
+    config: Dict[Text, Any], keys_to_configure: List[Text]
+) -> Dict[Text, Any]:
+    """Complete a config by adding automatic configuration for the specified keys.
+
+    Args:
+        config: The provided configuration.
+        keys_to_configure: Keys to be configured automatically (e.g. `policies`).
+
+    Returns:
+        The resulting configuration including both the provided and the automatically
+        configured keys.
+    """
+    import pkg_resources
+
+    if keys_to_configure:
+        logger.debug(
+            f"The provided configuration does not contain the key(s) "
+            f"{common_utils.transform_collection_to_sentence(keys_to_configure)}. "
+            f"Running automatic configuration for them now."
+        )
+
+    default_config_file = pkg_resources.resource_filename(
+        __name__, "default_config.yml"
```

Comment:
I've just discovered `cli/default_config`. Can we use that one? Or do we even need that one? Might be beyond the scope of this PR.

Reply:
`cli/default_config.yml` seems to be an old remainder that we don't need anymore.
It was introduced with https://github.com/RasaHQ/rasa/commit/d135027dfb07f0a2ad9800252d978be4b5dce13a and used in a constant `FALLBACK_CONFIG_PATH`, which has since been removed. I checked and it seems like the file is not used anywhere now.

For the auto configuration, it makes more sense to have the default config stored in `importers` than in `cli`, so let's get rid of the one in `cli`?

Reply:
Makes sense. However, some tests depend on it I think. So unless it's a quick thing, I'd create an issue for it

Reply:
Ah, I had somehow missed those tests. Re-added it for now, will create an issue.